api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -371,6 +380,18 @@ message HttpConnectionManager {
   ServerHeaderTransformation server_header_transformation = 34
       [(validate.rules).enum = {defined_only: true}];
 
+  // Defines the action to be applied to the Scheme header on the request path
+  // if scheme is set in the config. Defaults to OVERWRITE_SCHEME.
+  SchemeHeaderTransformation scheme_header_transformation = 48
+      [(validate.rules).enum = {defined_only: true}];
+
+  // An optional override. If set, the connection manager will set the request
+  // scheme to this value based on scheme_header_transformation. Valid values are 'http' and
+  // 'https'. If absent, Envoy will pass through downstream scheme if present or
+  // set scheme based on downstream encryption level.
+  string scheme = 49
+      [(validate.rules).string = {well_known_regex: HTTP_HEADER_VALUE strict: false}];
+
```

Comment:
We may need this on per-route base.  Will it be this PR?

Reply:
happy to add if needed, but is there a use case for it?

Reply:
Hmm,  I don't have one for now, at least for ESPv2.   We can add it when it is needed.

Reply:
I think I might recommend doing the following, both to make this API a bit cleaner and also to facilitate per-route overrides:
1) Define a `SchemeHeaderTransformation` message somewhere common, and refer to it here.
2) I would make the absence of setting this message the new default behavior.
3) Setting this message turns on override behavior. Perhaps initially we only need a `overwrite_scheme` field?

Also, IMO this topic is sufficiently confusing that I would recommend a) adding an arch overview section under HTTP around "scheme handling" and b) really beefing the docs up with examples on what each setting will do both with default behavior and with overrides.

WDYT?